src/OSPSuite.Core/Domain/Services/PKCalculationOptionsFactory.cs

```diff
@@ -27,14 +27,19 @@ public PKCalculationOptions CreateFor(IModelCoreSimulation simulation, string mo
          var applicationEndTimes = new List<float>(applicationStartTimes.Skip(1)) {endTime};
 
 
-         for (int i = 0; i < applicationStartTimes.Length; i++)
+         for (var i = 0; i < applicationStartTimes.Length; i++)
          {
             var dosingInterval = new DosingInterval
             {
                StartValue = applicationStartTimes[i],
                EndValue = applicationEndTimes[i]
             };
 
+            //as soon as one dosing interval is not valid, it means that the end time might be on exactly the start time of an application.
+            //Everything after this is by construction invalid so we do not consider further interval for the calculations
+            if (!dosingInterval.IsValid)
```

Comment:
@Yuri05 Here, I am checking that the dosing Interval is well defined (StartTime < EndTime). If not, it means that the EndTime of sim is before StarTTime (or equal). Everything after is also unusable